dcpy/connectors/edm/recipes.py

```diff
@@ -271,3 +324,73 @@ def purge_recipe_cache():
     """Delete locally stored recipe files."""
     logger.info(f"Purging local recipes from {LIBRARY_DEFAULT_PATH}")
     shutil.rmtree(LIBRARY_DEFAULT_PATH)
+
+
+def scrape_metadata(dataset: str) -> None:
+    """For a given recipes dataset, regenerates logs"""
+    logger.info(f"Re-scraping metadata for dataset {dataset}")
+    pg_client = postgres.PostgresClient(database=LOGGING_DB, schema=LOGGING_SCHEMA)
+
+    ## remove records in db for this recipe dataset
+    try:
+        pg_client.execute_query(
+            f"""
+            DELETE FROM {LOGGING_TABLE_NAME}
+            WHERE name = :dataset AND event_source='scrape';""",
+            dataset=dataset,
+        )
+    except Exception as e:
+        logger.warn(f"Could not delete from table: {e}")
+
+    logged_metadata = get_logged_metadata([dataset])
+    logged_versions = {m["version"] for _, m in logged_metadata.iterrows()}
+
+    ## scrape metadata, insert into table
+    versions = get_all_versions(dataset)
+    metadata = [
+        get_archival_metadata(dataset, v).to_row()
+        for v in versions
+        if v not in logged_versions
```

Comment:
Rationale for this being if we archive v 20240101 on 20240102, and then re-archive it on 20240103, edm-recipes only gives us the latest. But both events should now be logged when actually running `library` (see logging commit). If events are logged by data library and already in the table, they don't need to be re-scraped

Theoretically, after merging this and scraping all the data once, we won't need to do it ever again (but we know that's not true)